pkg/services/publicdashboards/database/database_test.go

```diff
@@ -710,6 +710,64 @@ func TestGetDashboardByFolder(t *testing.T) {
 	})
 }
 
+func TestGetMetrics(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+	var sqlStore db.DB
+	var cfg *setting.Cfg
+	var dashboardStore dashboards.Store
+	var publicdashboardStore *PublicDashboardStoreImpl
+	var savedDashboard *dashboards.Dashboard
+	var savedDashboard2 *dashboards.Dashboard
+	var savedDashboard3 *dashboards.Dashboard
+	var savedDashboard4 *dashboards.Dashboard
+
+	setup := func() {
+		sqlStore, cfg = db.InitTestDBwithCfg(t, db.InitTestDBOpt{FeatureFlags: []string{featuremgmt.FlagPublicDashboards}})
+		quotaService := quotatest.New(false, nil)
+		store, err := dashboardsDB.ProvideDashboardStore(sqlStore, cfg, featuremgmt.WithFeatures(), tagimpl.ProvideService(sqlStore, cfg), quotaService)
+		require.NoError(t, err)
+		dashboardStore = store
+		publicdashboardStore = ProvideStore(sqlStore)
+		savedDashboard = insertTestDashboard(t, dashboardStore, "testDashie", 1, 0, false)
+		savedDashboard2 = insertTestDashboard(t, dashboardStore, "testDashie2", 1, 0, false)
+		savedDashboard3 = insertTestDashboard(t, dashboardStore, "testDashie3", 2, 0, false)
+		savedDashboard4 = insertTestDashboard(t, dashboardStore, "testDashie4", 2, 0, false)
+		insertPublicDashboard(t, publicdashboardStore, savedDashboard.UID, savedDashboard.OrgID, true, PublicShareType)
+		insertPublicDashboard(t, publicdashboardStore, savedDashboard2.UID, savedDashboard2.OrgID, true, PublicShareType)
+		insertPublicDashboard(t, publicdashboardStore, savedDashboard3.UID, savedDashboard3.OrgID, true, EmailShareType)
+		insertPublicDashboard(t, publicdashboardStore, savedDashboard4.UID, savedDashboard4.OrgID, false, EmailShareType)
+	}
+
+	t.Run("returns correct list of metrics", func(t *testing.T) {
+		setup()
+
+		metrics, err := publicdashboardStore.GetMetrics(context.Background())
+		require.NoError(t, err)
+
+		assert.Equal(t, 3, len(metrics.TotalPublicDashboards))
+		enabledAndPublicCount := 0
+		enabledAndEmailCount := 0
+		disabledAndEmailCount := 0
```

Comment:
I would initialize this with a value like `-1` to be sure if there is a 0 it's because we read it from the DB instead of the default value.